Add Batch dataplane API version 2018-12-01.8.0

matthchr opened the pull request:
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.

This is a data plane API so ARM review is not required.

I tried to run the linter here: https://portal.azure-devex-tools.com/app/tools/linter but it just hung on "still working!" forever. 
I also tried to run this: https://portal.azure-devex-tools.com/app/tools/static-validation/static/errors/default pointing at `https://github.com/matthchr/azure-rest-api-specs/tree/feature/batch-dataplane-2018-12/specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0` and I also got an error "Sorry something went wrong!"

I did run the azure validator manually though and things looked okay (there's some warnings but they basically don't seem to apply to data-plane)

xingwu1 committed:
Prepare for Azure Batch new API version

xingwu1 committed:
Add new data plane API version 2018-12-01.8.0

matthchr committed:
Fix documentation issues

AutorestCI commented:
# Automation for azure-sdk-for-js

AutorestCI commented:
# Automation for azure-sdk-for-ruby

AutorestCI commented:
# Automation for azure-sdk-for-python

AutorestCI commented:
# Automation for azure-sdk-for-java

AutorestCI commented:
# Automation for azure-sdk-for-node

azuresdkci commented:
Can one of the admins verify this patch?

AutorestCI commented:
# Automation for azure-sdk-for-go

anuchandy commented:
@matthchr I will talk to owners of the devex tool on the issues you faced.

Could you take a look at model (example) validation https://travis-ci.org/Azure/azure-rest-api-specs/jobs/463987019?

matthchr commented:
@anuchandy - I will fix the example validation issues and update the PR

matthchr committed:
Fix examples

 - Add batchUrl parameter.
 - Rename "ocp-data" to "ocp-date" to correctly reflect what the API
   actually requires.

matthchr commented:
I fixed the original issue with the model validator, but am now getting a different one which seems very unclear to me. See: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/464004216

```
operationId: Application_Get
scenario: Get applications
source: request
responseCode: ALL
severity: 0
errorCode: REQUIRED
errorDetails:
  code: REQUIRED
  message: Value is required but was not provided
  path: >-
    /applications/{applicationId}/applicationId/paths//applications/{applicationId}/get/parameters/0
```

It's not clear to me what the actual problem is...

dlepow commented on specification/batch/data-plane/Microsoft.Batch/stable/2018-12-01.8.0/BatchService.json:
```diff
@@ -0,0 +1,15868 @@
+            "required": true,
+            "type": "string",
+            "description": "The path to the task file or directory that you want to delete."
+          },
```
@matthchr - Matt, sorry, can you please update this description to change the incorrect doc link? I suggest: 
 "description": "This property is mutually exclusive with other ImageReference properties. The virtual machine image must be in the same region and subscription as the Azure Batch account. For more details, see https://docs.microsoft.com/azure/batch/batch-custom-images."

matthchr replied:
@dlepow did you put this on the wrong line?

matthchr replied:
I think I found where it was meant to be though -- I've fixed the docs there

dlepow replied:
@matthchr - Thanks. When I commented before the GitHub UI was a bit flaky with the large file, so I might have selected the wrong place.

matthchr commented:
@anuchandy I have confirmed that the URLs should contain HTTPS (at least, in the code generated by AutoRest you must set the URL to `https://<account>.<region>.batch.azure.com`). Given that, I think it makes sense to leave the examples with the HTTPS, as I am pretty sure it's supposed to be there.

Is there anything else that needs to be fixed, or is this good to merge?

anuchandy commented:
@sergey-shandar - do you think this is a bug in OAV? OR should "https" has to be removed from examples? we may want to set a guideline for service team here, what I understand from our hallway talk is - "the swagger scheme part says it's https and as per http convention, host name does not contain protocol." Can you please confirm?

Tracked here: https://github.com/Azure/oav/issues/368

matthchr commented:
@anuchandy - I believe the examples should be fixed now, I removed HTTPS from them
